flexiblepricing/mail_api.py

```diff
@@ -41,7 +41,7 @@ def generate_flexible_price_email(flexible_price):
         courserun = CourseRun.objects.filter(
             course=flexible_price.courseware_object
         ).first()
-        product = Product.objects.get(object_id=courserun.id)
+        product = Product.objects.get(object_id=courserun.id, is_active=True)
```

Comment:
I was just thinking, Instead of getting the products with `is_active` filter in different places, Would it be good that we change the manager in a way that whenever `Product.objects` manager is called it returns the active products only? I created [that manager](https://github.com/mitodl/mitxonline/blob/main/ecommerce/models.py#L45-L51) recently with similar intent:
so if we replace
`
       return ProductsQuerySet(self.model, using=self._db)
`

with

`
        return ProductsQuerySet(self.model, using=self._db).filter(is_active=True)
`

Calling the `Product.objects` will always return Active product everywhere in the existing code, and if we want all products we could use [Product.all_objects](https://github.com/mitodl/mitxonline/blob/main/ecommerce/models.py#L79).

Reply:
I think we'd want to be very careful here with that kind of change - it'd be very easy to introduce a regression - I also think it warrants a separate PR.

Reply:
@rhysyngsun you're correct.